src/main/java/RandomGenerator.java

```diff
@@ -202,8 +202,19 @@ private Map<Object, Object> getRandomMap(Type type) {
     }
 
     public <T> List<T> getRandomList(Type type) {
-        // TODO @ibrahim
-        return null;
+        int randomSize = ThreadLocalRandom
+                .current()
+                .nextInt(MIN_COLLECTION_SIZE, MAX_COLLECTION_SIZE) ;
+        ParameterizedType parameterizedType = (ParameterizedType)type ;
+        Type inputListType = parameterizedType.getActualTypeArguments()[0] ;
+        // Class<?> inputListClass = (Class<?>) parameterizedType.getActualTypeArguments()[0];
+
+        List<Object> outputList = new ArrayList<Object>() ;
```

Comment:
it would be better in terms of performance to initialize the outputList's size : 
```java
        List<Object> outputList = new ArrayList<Object>(randomSize) ;
```

Reply:
YES Agree, Thanks for the review I 'll change it